parsers/PA.py

```diff
@@ -174,9 +180,88 @@ def fetch_production(zone_key='PA', session=None, target_datetime=None, logger:
 
     return data
 
+def fetch_exchange(zone_key1='CR', zone_key2='PA', session=None,
+                   target_datetime=None, logger=logging.getLogger(__name__)) -> dict:
+    """
+    Requests the last known power exchange (in MW) between two countries.
+    """
+
+    if target_datetime:
+        raise NotImplementedError("This parser is not yet able to parse past dates")
+    
+    sorted_zone_keys = '->'.join(sorted([zone_key1, zone_key2]))
+
+    r = session or requests.session()
+    url = EXCHANGE_URL
+
+    response = r.get(url)
+    assert response.status_code == 200
+
+    df = pd.read_html(response.text)[0]
+
+    # A positive value on website indicates a flow from country specified to PA.
+    net_flow_cr = round(float(df[4][1]) + float(df[4][3]) + float(df[4][5]) + float(df[1][8]) + float(df[1][10]), 2)
+    net_flow_gt = round(float(df[4][23]) + float(df[4][26]) + float(df[4][28]) + float(df[1][31]), 2)
+    net_flow_hn = round(float(df[1][13]) + float(df[1][15]) + float(df[1][18]) + float(df[1][20]) + float(df[1][23]), 2)
+    net_flow_ni = round(float(df[4][8]) + float(df[4][10]) + float(df[4][13]) + float(df[4][15]), 2)
+
+    # invert sign to account for direction in alphabetical order
+    net_flow_sv = -1 * round(float(df[4][18]) + float(df[4][20]) + float(df[1][26]) + float(df[1][28]), 2)
+    
+    net_flows = {
+      "CR->PA": net_flow_cr, # Costa Rica to Panama
+      "GT->PA": net_flow_gt, # Guatemala to Panama
+      "HN->PA": net_flow_hn, # Honduras to Panama
+      "NI->PA": net_flow_ni, # Nicaragua to Panama
+      "PA->SV": net_flow_sv, # Panama to El Salvador
+    }
+
+    if sorted_zone_keys not in net_flows:
+        raise NotImplementedError(f"This exchange pair is not implemented: {sorted_zone_keys}")
+    
+    data = {
+        'datetime': arrow.now(TIMEZONE).datetime,
+        'netFlow': net_flows[sorted_zone_keys],
+        'sortedZoneKeys': sorted_zone_keys,
+        'source': url
+    }
+
+    return data
+
+def fetch_consumption(
+    zone_key="PA", session=None, target_datetime=None, logger=logging.getLogger(__name__)) -> dict:
+    """
+      Fetches consumption of Panama.
+    """
+
+    if target_datetime:
+        raise NotImplementedError("This parser is not yet able to parse past dates")
+
+    r = session or requests.session()
+    url = CONSUMPTION_URL
+
+    response = r.get(url)
+    assert response.status_code == 200
+
+    soup = BeautifulSoup(response.text, 'html.parser')
+    consumption_div = soup.find_all('div', {'class': 'widget-small info coloured-icon'})[0]
```

Comment:
I think this part of using the classes used for presentational values is a bit fragile - if they decide to change color of the box, this will break the parser. Could we use the `h5` title of "Demanda Total" to locate the relevant place instead? 

Think something like this could work, but maybe it can be done in a better way? :)

```python
    consumption_title = soup.find("h5", string=re.compile(r"\s*Demanda Total\s*"))
    consumption_val = consumption_title.find_next_sibling().text.split()[0]
```

Reply:
Yes indeed, that would likely improve the resilience of that parser! Thank you for highlighting!